### PR TITLE
Map nodeID to parents

### PR DESCRIPTION
**Pull request by leenlab2**
Created an object mapping nodeIDs to nodeIDs/boolIDs of the nodes preceding the current one (parent nodes). This includes:

- Added a new `parents` state to `Graph`
- Initialized an object `parentsObj` with the `node.id_` as the keys and a blank list for values by iterating over `nodesList`
- Updated the object by iterating over the `edgesList`
- Added a comment explaining why this object was created
- Set `parents` to `parentsObj`

### CODE REVIEW

**Comment by leenlab2** [comment]
Update:

- Removed the old `parents` state from Graph
- Added a new `parentTree` state where `parentTree.parents` refers to the old object, and `parentTree.inEdges` refers to a similar object that holds edgeIDs
- Passed `parentTree` down to NodeGroup as a prop
- Passed `parentTree.parents` and `parentTree.inEdges` down to Node as props
- Removed previous code that created these props
- Updated NodeGroup PropTypes

File: js/components/graph/Graph.js [hunk 1]
```diff
@@ -159,6 +162,20 @@ export default class Graph extends React.Component {
             edgesList.push(entry);
           }
         });
+
+        // An object mapping a node to its parents/childs improves efficiency in NodeGroup/BoolGroup
```
Comment (david-yz-liu) [inline js/components/graph/Graph.js:166]: Remove this comment. The term "improves" implies a comparison or change that won't make sense once the old code is deleted. This is a situation where the "why" is best described in the pull request description.

**Review by david-yz-liu — COMMENTED** [review]
@leenlab2 great work! I left one comment. Also, you will need to resolve a merge conflict with master. This one should be pretty simple to solve: do a pull from master, and then look for the conflict markers `>>>>>>>`/`<<<<<<<<` , which mark the sections in the conflict. You can talk to your team members too.

Ref: https://docs.github.com/en/github/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

**Comment by leenlab2** [comment]
Update:

- I removed the comment like you asked
- I resolved the conflict with master

**Pull request merged**